### ISSUE TITLE
Orchestration - mgDiagSettingsAll error - InvalidAuthenticationToken

### ISSUE THREAD

**Issue by baartch** [Azure/ALZ-Bicep#470]
## Describe the bug

Im running into exactly the same issue as #465, at step 4.1 of the Deployment Workflow.

## To Reproduce

- I followed every step like documented in the workflow
- At all the previous steps, I choosed default values whenever possible
- I have only one single Subscription. I manually moved it to the _Management_ Management Group (but tried it on the root level as well)
- The same errors appear on my local machine and in the Cloud Powershell
- I tried the Powershell and the Azure CLI command
- All deployments and resources went into the region _switzerlandnorth_. (No problems so far)

I provided a minimal Parameter file: `mgDiagSettingsAll.parameters.min.json`
```json
{
  "$schema": "https://schema.management.azure.com/schemas/2019-04-01/deploymentParameters.json#",
  "contentVersion": "1.0.0.0",
  "parameters": {
    "parTopLevelManagementGroupPrefix": {
      "value": "clxs"
    },
    "parLogAnalyticsWorkspaceResourceId": {
      "value": "/subscriptions/e4456dc5-d177-4b75-b3db-deece2d26ef2/resourceGroups/rg-clxs-logging-001/providers/Microsoft.OperationalInsights/workspaces/clxs-log-analytics"
    },
    "parTelemetryOptOut": {
      "value": false
    }
  }
}
```

## Screenshots 📷

![image](https://user-images.githubusercontent.com/26668031/223400512-bc5c0ce2-b6bf-49c2-b40f-cf5fdd2edfb1.png)

## Correlation ID

Correlation ID : f6955b66-1daa-4400-b710-cf08a4cf1ae8

Thank you very much for your help.

**Comment by jtracey93** [comment]
Hey @baartch,

Can you confirm you are deploying the Management Groups via the ALZ Bicep module and are you putting a dependency on that completing before trying to deploy the MG diag settings?

**Comment by baartch** [comment]
Hi @jtracey93 
Yes, I deployed the Management Groups with the ALZ Bicep module (Step 1 of the Deployment Flow). I did this already two weeks ago, so this is defenitely completed.
Thanks for the fast reply.

**Comment by jtracey93** [comment]
No worries, and thanks for coming back @baartch ?

And re-running this doesnt make any difference? You definitely updated the parameters file for mgDiagSettingsAll, right for the MG prefix?

**Comment by baartch** [comment]
I retried it several times, with PS and AzCLI, locally and Cloud Shell.

Here are my Management Groups (with prefix _clxs_, like I set in the parameters file)
![image](https://user-images.githubusercontent.com/26668031/223408056-e7332261-93ad-47a4-80a7-9ed9d468ceb8.png)

**Comment by jtracey93** [comment]
Thanks @baartch and di you set this in the mgDiagSettings.bicep paramaters file for the `clxs` prefix? It looks like it from your original screenshot but just confirming.

Can you also confirm the following for me please:

1. Is the subscription containing the log analytics workspace a new one or an existing one?
2. Is the `microsoft.insights` Resource Provider registered on the subscription? `Get-AzResourceProvider -ProviderNamespace 'Microsoft.Insights' | FT`

Reason for asking is ive just opened PR #471 to trigger our test framework which deploys our modules, and this went straight through no issues at all:

![image](https://user-images.githubusercontent.com/41163455/223411730-d56310bf-c64a-42b7-bc8e-0a691196307f.png)

> Pipeline used can be seen here: https://github.com/Azure/ALZ-Bicep/blob/main/tests/pipelines/bicep-build-to-validate.yml

![image](https://user-images.githubusercontent.com/41163455/223411929-3891b24d-17a9-46fa-be93-c61c34aa3495.png)

![image](https://user-images.githubusercontent.com/41163455/223412263-bad92828-36b6-4a89-8ef1-0fac485c896c.png)

This is on a freshly created subscription and we register the resource provider manually also before we do the deployments, as you can see in the above pipeline.

So trying to work out the differences here as im unable to repro currently.

Are you able to open a support case so support can investigate, as I dont think this is an ALZ Bicep modules issue and potentially instead a platform issue. If you can please share the support case ID with me so I can track.

Thanks & Let me know

Jack

**Comment by jtracey93** [comment]
Okay @baartch & @cyb3n3tic, I have been able to repro in #472 by note registering the resource provider automatically for `Microsoft.Insights` on a newly created subscription.
![image](https://user-images.githubusercontent.com/41163455/223440977-998bc00f-35e5-444f-bdd3-ac1e773a4c18.png)

And on the intermediate root management group deployment i am seeing the error of the RP not being registered, as shown below:
![image](https://user-images.githubusercontent.com/41163455/223441197-3d9d98a1-8c2b-441e-9bb4-a5db3a88c677.png)

Are either, or both, of you able to confirm that the subscription that hosts the Log Analytics Workspace has the `Microsoft.Insights` resource provider registered? You can run `Get-AzResourceProvider -ProviderNamespace 'Microsoft.Insights' | FT` to check.

If it isn't, can you run `Register-AzResourceProvider -ProviderNamespace 'Microsoft.Insights'` and wait for it to succeed and finish and then re-run the `mgDiagSettingsAll.bicep` deployment and let me know the results?

Thanks

Jack

**Comment by cyb3n3tic** [comment]
Hi Jack,

I have not looked at Landing Zones since deploying that demo last time and have since torn it down however I have just hopped on to the same tenant with context set to the management subscription and ran the check above which is showing RegistrationState of Registered for all.

I might have time at some point to try another demo, likely to be the week after next though.

Cheers.

Ben

**Comment by baartch** [comment]
@jtracey93 You are great!

`Register-AzResourceProvider -ProviderNamespace 'Microsoft.Insights'` solved the problem. I have a brand new test Subscription and by now, I was not aware I have to register specific ResourceProviders.

But interestgingly, my tenant didn't come up with an error message like yours.
![image](https://user-images.githubusercontent.com/26668031/223634166-f78b7b86-f86a-4044-93c8-0458acc25a7d.png)

After registering, I waited until the command `Get-AzResourceProvider -ProviderNamespace 'Microsoft.Insights' | FT` told me on every single line _microsoft.insights Registered_.
On the first run, not everything went smooth. Some were green, some were red. I just fired the command a second time and then the deployment went through.
![image](https://user-images.githubusercontent.com/26668031/223634782-008d3111-2162-477d-85b3-ca8352119b37.png)

Now, everything is green and I'm happy to go on with the _Deployment Flow_.
Maybe this has to go into the Wiki?

Thank you Jack for your very fast help. You're doing an awesome job!
